tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.IdentityModel.Abstractions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.PublicApiTests
+{
+    [TestClass]
+    public class TelemetryClientTests : TestBase
+    {
+        [TestMethod]
+        public void TelemetryClientExperimental()
+        {
+            ITelemetryClient telemetryClient = new TelemetryClient(TestConstants.ClientId);
+
+            var e = AssertException.Throws<MsalClientException>(() => ConfidentialClientApplicationBuilder
+                .Create(TestConstants.ClientId)
+                .WithClientSecret("secret")
+                .WithTelemetryClient(telemetryClient)
+                .Build());
+
+            Assert.AreEqual(MsalError.ExperimentalFeature, e.ErrorCode);
+        }
+    }
+
+    internal class TelemetryClient : ITelemetryClient
```

Comment:
nit: > something like `TestTelemetryClient` or you can just pass a mocked `ITelemetryClient`.

Reply:
Agreed. It doesn't seem like you are doing very complicated stuff with this implementation, so using `Substitute.For<ITelemetryClient>` is probably enough.

Reply:
Does MSAL leverage Moq at all? seems like that could be useful here.

Reply:
We use NSubstitute.

Reply:
I am adding implementation to this in the next PR to test what is logged. So leaving this as is.